include/stumpless/error.h

```diff
@@ -30,85 +30,100 @@
  *
  * @since release v1.6.0.
- */
-  STUMPLESS_DUPLICATE_ELEMENT,
+ */\
+  ERROR(STUMPLESS_DUPLICATE_ELEMENT, 3)	\
 /**
  * No element with the specified characteristics was found.
  *
  * @since release v1.6.0.
- */
-  STUMPLESS_ELEMENT_NOT_FOUND,
-/** Could not open a file. */
-  STUMPLESS_FILE_OPEN_FAILURE,
-/** Could not write to a file. */
-  STUMPLESS_FILE_WRITE_FAILURE,
-/** A provided index was not in a valid range. */
-  STUMPLESS_INDEX_OUT_OF_BOUNDS,
-/** A provided facility code did not conform to RFC 5424. */
-  STUMPLESS_INVALID_FACILITY,
-/** A target or entry ID was not valid. */
-  STUMPLESS_INVALID_ID,
+ */\
+  ERROR(STUMPLESS_ELEMENT_NOT_FOUND, 4)	\
+/** Could not open a file. */\
+  ERROR(STUMPLESS_FILE_OPEN_FAILURE, 5)	\
+/** Could not write to a file. */\
+  ERROR(STUMPLESS_FILE_WRITE_FAILURE, 6)	\
+/** A provided index was not in a valid range. */\
+  ERROR(STUMPLESS_INDEX_OUT_OF_BOUNDS, 7)	\
+/** A provided facility code did not conform to RFC 5424. */\
+  ERROR(STUMPLESS_INVALID_FACILITY, 8)	\
+/** A target or entry ID was not valid. */\
+  ERROR(STUMPLESS_INVALID_ID, 9)	\
 /**
  * A provided severity code did not conform to RFC 5424.
  *
  * @since release v1.6.0.
- */
-  STUMPLESS_INVALID_SEVERITY,
-/** A memory allocation or reallocation failed. */
-  STUMPLESS_MEMORY_ALLOCATION_FAILURE,
-/** The given layer 3 protocol is not supported by this implementation. */
-  STUMPLESS_NETWORK_PROTOCOL_UNSUPPORTED,
+ */\
+  ERROR(STUMPLESS_INVALID_SEVERITY, 10)	\
+/** A memory allocation or reallocation failed. */\
+  ERROR(STUMPLESS_MEMORY_ALLOCATION_FAILURE, 11)	\
+/** The given layer 3 protocol is not supported by this implementation. */\
+  ERROR(STUMPLESS_NETWORK_PROTOCOL_UNSUPPORTED, 12)	\
 /**
  * No param with the specified characteristics was found.
  *
  * @since release v1.6.0.
+ */\
+  ERROR(STUMPLESS_PARAM_NOT_FOUND, 13)	\
+/** Could not bind to a socket. */\
+  ERROR(STUMPLESS_SOCKET_BIND_FAILURE, 14)	\
+/** Could not connect to a socket. */\
+  ERROR(STUMPLESS_SOCKET_CONNECT_FAILURE, 15)	\
+/** Could not create a socket. */\
+  ERROR(STUMPLESS_SOCKET_FAILURE, 16)	\
+/** Could not send to a socket. */\
+  ERROR(STUMPLESS_SOCKET_SEND_FAILURE, 17)	\
+/** Could not write to a stream. */\
+  ERROR(STUMPLESS_STREAM_WRITE_FAILURE, 18)	\
+/** The requested operation is not valid on the given target type. */\
+  ERROR(STUMPLESS_TARGET_INCOMPATIBLE, 19)	\
+/** The target type provided is unsupported by this implementation. */\
+  ERROR(STUMPLESS_TARGET_UNSUPPORTED, 20)	\
+/** The given layer 4 protocol is not supported by this implementation. */\
+  ERROR(STUMPLESS_TRANSPORT_PROTOCOL_UNSUPPORTED, 21)	\
+/** Could not close the Windows Event Log. */\
+  ERROR(STUMPLESS_WINDOWS_EVENT_LOG_CLOSE_FAILURE, 22)	\
+/** Could not open the Windows Event Log. */\
+  ERROR(STUMPLESS_WINDOWS_EVENT_LOG_OPEN_FAILURE, 23)	\
+
+#define GENERATE_ENUM(ENUM, INDEX) ENUM = INDEX,
```

Comment:
rename this to `STUMPLESS_GENERATE_ENUM` so that it does not interfere with the namespace of any files that include this header